compiler/src/iree/compiler/Dialect/HAL/Target/LLVMCPU/LLVMCPUTarget.cpp

```diff
@@ -669,6 +669,10 @@ class LLVMCPUTargetBackend final : public TargetBackend {
         return variantOp.emitError()
                << "failed to compile LLVM-IR module to an object file";
       }
+      if (!options.dumpIntermediatesPath.empty()) {
+        dumpDataToPath(options.dumpIntermediatesPath, options.dumpBaseName,
+                       variantOp.getName(), ".o", objectData);
```

Comment:
alllll the docs nits today: this could be reflected here: https://github.com/openxla/iree/blob/cbe57993986207547307497b6d4e1558557308a1/docs/website/docs/developers/general/developer-tips.md?plain=1#L166-L176

https://iree.dev/developers/general/developer-tips/#dumping-executable-files